packages/botonic-nlp/src/codify/codifier.ts

```diff
@@ -0,0 +1,39 @@
+import { unique } from '../utils/array-utils'
+
+export class Codifier {
+  vocabulary: string[]
+
+  constructor(vocabulary: string[]) {
+    this.vocabulary = unique(vocabulary)
+  }
+
+  encode(sequence: string[]): number[] {
+    return sequence.map(token => this.encodeToken(token))
+  }
+
+  private encodeToken(token: string): number {
+    if (!this.isValidToken(token)) {
```

Comment:
this has 2 issues:
1) performance: you're finding the token in the vocabulary twice
2) simplicity: isValidToken function would make sense if you needed it as public function or if you called it from more than 1 place. Otherwise, you're just introducing complexity by separating the check from the exception